### ISSUE TITLE
[BUG] Wrong memory unit

### ISSUE THREAD

**Issue by mohi001** [aristocratos/btop#283]
Unit of memory in proc section is wrong

**To Reproduce**

Call floating_humanizer() with a size less than 10 when shortened and slightly less than a integer when converted to MB
 e.g(8388608 - 10)
this will result 8G
expected
8.0M

![Sc_14001216_20220307_15:51:40](https://user-images.githubusercontent.com/72943169/157033661-af880554-5928-4b36-8e7f-f49f4b84e2c7.png)

 btop++ version:1.2.5
 Commit `2fa800e21e2a74f3b12f52178a75e067ce349648` caused this

**Comment by arrowgent** [comment]
just came to report the same but only for bpytop++ 
glad to see on the release page it was updated and fixed

3.9M
4G
4.0M (not sure why this happened occasionally)

![Screenshot at 2022-04-25 05-22-38](https://user-images.githubusercontent.com/57545549/165089008-4989841d-a486-4511-82ff-96f0c2d38208.png)
![Screenshot at 2022-04-24 22-43-27](https://user-images.githubusercontent.com/57545549/165089011-70118ddf-363c-4309-ad9e-e3a39475aead.png)

**Comment by sehe** [comment]
I too was going to submit a fix. I'd like to note that the problem was much broader than the title suggests. I notices because my processes routinely mentioned using e.g. 6T for no reason.

I had these test cases:

```c++
auto test = [](uint64_t v) {
    std::cout << v << "\t" << floating_humanizer(v, false) << "\t" << floating_humanizer(v, true) << "\t" << std::endl;
};
for(uint64_t i : {0, 1, 2, 3, 4, 5, 6, 7, 8, 9})
{
    for(uint64_t s : {
            1,                  // B
            1024,               // K
            1024 * 1024,        // M
            1024 * 1024 * 1024, // G
        })
    {
        auto v = s * i;
        if(v)
            test(v - 1);
        test(v);
        test(v + 1);
    }
}
```

Out of 116 output lines, 48 (!!) were wrong.

The fix #284 does fix them:

![image](https://github.com/aristocratos/btop/assets/324097/e8691144-addd-49f1-9126-c27f6b50b2d0)
